### PR TITLE
add episode 005 transcript

### PR DESCRIPTION
**Pull request by ericseppanen**

### CODE REVIEW

File: _transcripts/005-rust-1.38.0.md [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+
+__Jon__: Yeah, you can think of this as, you don't need the assembly code for
+all of the crates you're depending on, in order to compile your crate. Because
+much of it will be function calls into some part of the code that is external to
+the current crate. And then this is where link time optimization comes in. Where
+at link time, you can then try to do optimization across crate boundaries.
+
+The other thing, that's a little neat, as we talked about back in 1.36, the "mem
+uninit" type landed. And this was to try to deal with some issues around
+incorrect use of `mem::uninitialized`.
```
Comment (ericseppanen) [inline _transcripts/005-rust-1.38.0.md:81]: I'm not sure what this talk of "mem uninit" is.  Seems like this is talking about MaybeUninit?  If "mem uninit" is wrong, I'm not sure if it should be formatted as code.

File: _transcripts/005-rust-1.38.0.md [hunk 2]
```diff
@@ -0,0 +1,686 @@
...
+__Jon__: And it's because it means that semantically, semver-compatible updates
+of dependencies might now break your build, even though the update was, in fact,
+completely semver-compatible.
+
+There's another thing that landed in 1.38 which is this `std::any::type_name`.
+So this is primarily used in debug context. So the idea here is that it's
+generic over some type `T` just like the `size_of` function, for example, and it
+just gives you a string that describes the type you're talking about. If you run
+`type_name`, generic over `bool`, then you'll get back a string that just says
+`"bool"`. And the idea is, you can use this to print out your types, if you
```
Comment (ericseppanen) [inline _transcripts/005-rust-1.38.0.md:237]: Perhaps this should be formatted as `type_name::<bool>` ?

File: _transcripts/005-rust-1.38.0.md [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+know about this method from before, but if you have a slice, or anything that
+derefs to a slice, then you can join them with some separators. So the idea here
+is— the example that comes most often to mind is, you have a string or you have
+a slice of strings and you want to join them by some separator. So this is
+usually something like a comma. You want a list to be comma separated as a
+single string, and you can do the same thing for a slice. You have a slice of
+`u8`s and you want to join those many slices of `u8`s by some separator, and
+previously you could only join them by a single separator, so a single `u8`.
+Whereas now, with this new change, you can join by something that is a slice
+itself, so you might join by something like `[0, 1, 2]`.
```
Comment (ericseppanen) [inline _transcripts/005-rust-1.38.0.md:337]: I'm not certain that `[0, 1, 2]` is what was meant here.

File: _transcripts/005-rust-1.38.0.md (hunk 1 above)
Comment (jonhoo) [reply _transcripts/005-rust-1.38.0.md:81]: Yup, definitely intended to be `MaybeUninit` (which indeed landed in 1.36). Not sure whether the transcript should have the error or the correct version?

File: _transcripts/005-rust-1.38.0.md (hunk 2 above)
Comment (jonhoo) [reply _transcripts/005-rust-1.38.0.md:237]: No, I think sticking to how it was phrased is fine here.

File: _transcripts/005-rust-1.38.0.md (hunk 3 above)
Comment (jonhoo) [reply _transcripts/005-rust-1.38.0.md:337]: I believe the intention here was that you can now join with a list of values which will be used in sequence. So I think I'd just typeset this as a sequence: `0, 1, 2`

File: _transcripts/005-rust-1.38.0.md (hunk 1 above)
Comment (ericseppanen) [reply _transcripts/005-rust-1.38.0.md:81]: I don't have any other ideas, unless we'd like to sort of break the 4th wall and insert:

_(editor's note: should be `MaybeUninit`)_

Your call.

Comment (jonhoo) [reply _transcripts/005-rust-1.38.0.md:81]: I like the idea of an editor's note.

Comment (ericseppanen) [reply _transcripts/005-rust-1.38.0.md]: OK, done.

File: _transcripts/005-rust-1.38.0.md (hunk 3 above)
Comment (ericseppanen) [reply _transcripts/005-rust-1.38.0.md]: OK, done.

**Review by jonhoo — APPROVED** [review]

**Pull request merged**